### PR TITLE
updates to zsh completion

### PR DESCRIPTION
**Pull request by pseyfert**
* update arguments to `catkin config` (from the help)
 * moved `_retrieve_cache` out of parentheses such that the call has an effect (see linked stackexchange entry)
 * updated the handling of multi-word arguments `--whitelist --blacklist --cmake-args --make-args --catkin-args` (detect if they got closed. if not complete them, if so go back to the regular `catkin config` completion and manipulate `words` and `CURRENT` to hide the args from `_arguments`)
 * suggest arguments that close multi word arguments when completing multi word arguments. (I suggest using this with something along the lines of `zstyle ':completion:*:descriptions' format '%d'`)
 * dispatch to the tab completion of cmake or make for the completion of `catkin config --(c)make-args`.
 * I picked the `path`→`_path` and `this_package`→`enclosing_package` fixes from https://github.com/catkin/catkin_tools/pull/482

### CODE REVIEW

**Comment by mikepurvis** [comment]
I'm not a zsh user, but I'm fine to merge this if you assert that it's good.

File: completion/_catkin [hunk 1]
```diff
@@ -276,23 +265,44 @@ _catkin_config_complete() {
...
-    '(--default-build-space)'{-b,--build-space}'[Build space path]'\
+    '(--default-source-space)'{-s,--source-space}'[Source space path]:source space:_files'\
+    '(-s --source-space)--default-source-space[Use the default path to the source space]'\
     '(--default-devel-space)'{-d,--devel-space}'[Devel space path]'\
+    '(-d --devel-space)--default-devel-space[Use the default path to the devel space]'\
+    '(--default-build-space)'{-b,--build-space}'[Build space path]'\
+    '(-b --build-space)--default-build-space[Use the default path to the build space]'\
     '(--default-install-space)'{-i,--install-space}'[Install space path]'\
+    '(-i --install-space)--default-install-space[Use the default path to the install space]'\
     {-x,--space-suffix}'[Suffix for build, devel, and install spaces]:suffix:()'\
```
Comment (timonegk) [inline completion/_catkin:286]: `-L`/`--log-space` seems to be missing here

File: completion/_catkin [hunk 2]
```diff
@@ -276,23 +265,44 @@ _catkin_config_complete() {
...
-    '--isolate-devel[Isolate devel speaces for each package]')
+    '(--link-devel --isolate-devel)--merge-devel[Build products directly into a merged devel space]'\
+    '(--merge-devel --isolate-devel)--link-devel[Symbolically link products into a merged devel space]'\
+    '(--merge-devel --link-devel)--isolate-devel[Isolate devel speaces for each package]'\
+    '(--no-install)--install[Causes each package to be installed to the install space]'\
+    '(--install)--no-install[Disables installing each package into the install space]'\
+    '(--merge-install)--isolate-install[Install each catkin package into a separate install space]'\
+    '(--isolate-install)--merge-install[Install each catkin package into a single merged install space]'\
+    {-j,--jobs}'[Maximum number of build jobs to be distributed across active packages]:JOBS:()'\
+    {-p,--parallel-packages}'[Maximum number of packages allowed to be built in parallel]:PACKAGE_JOBS:()'\
```
Comment (timonegk) [inline completion/_catkin:295]: `-l`/`--load-average` is missing (it was just added on master)

**Comment by spurnvoj** [comment]
@timonegk I made update according to your suggestions (see https://github.com/pseyfert/catkin_tools/pull/1). Can you please check it out?

**Comment by timonegk** [comment]
@spurnvoj I added your changes and rebased on master, everything works for me, if you could test once more and give a final okay, I will merge this.

**Comment by spurnvoj** [comment]
@timonegk works for me too.

**Pull request merged**